### PR TITLE
Sort events by title for selector

### PR DESCRIPTION
**Pull request by jessouyangbraven**
### Description

Sort the list of `CourseAttendanceEvent`s on the **Take Attendance** page by title.

### Motivation and Context

If we use the default order of `course.course_attendance_events`, we get the events sorted by `id`. This caused some confusion in Tuesday's demo of LCs taking attendance because the 1-1 attendance events appeared in the middle of Learning Lab attendance events.

This PR sorts the events alphabetically. Eventually, we might sort them, which approximates the Canvas assignment's `due_at` date.

### Screenshots (if appropriate):

- This is what we saw in production/during the demo, note that `TODO` appears in the middle of the list:
Screen Shot 2021-01-27 at 10 56 59 AM

- We don't bother alphabetizing the `course#edit` page, so the events appear in the order they were created:
Screen Shot 2021-01-27 at 10 52 53 AM

- With this PR in the LC Playbook's **Take Attendance**, the events are alphabetized:
Screen Shot 2021-01-27 at 10 52 37 AM

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist:

- [ ] My code follows the code style of this project.
- [ ] I have added neccessary labels
- [ ] My PR title follows the convention in the guide
- [ ] I have assigned myself to the PR
- [ ] I have requested reviews from team members
- [ ] I have created a task for reviewers on PM tool
- [ ] All new and existing tests passed.

### CODE REVIEW

File: app/models/course_attendance_event.rb [hunk 1]
```diff
@@ -9,6 +9,10 @@ class CourseAttendanceEvent < ApplicationRecord
 
   validates :course, :attendance_event, :canvas_assignment_id, presence: true
 
+  scope :order_by_title, -> {
+    joins(:attendance_event).order('attendance_events.title ASC')
```
Comment (rshipp) [inline app/models/course_attendance_event.rb:13]: will this sort "Learning Lab 2" after "Learning Lab 10" or before?

Comment (jessouyangbraven) [reply app/models/course_attendance_event.rb:13]: Ugh, worse. `10` is before `1` when using string comparison. :(

I guess there's no easy workaround except to use `due_at`.

Screen Shot 2021-01-27 at 2 29 18 PM

**Comment by rshipp** [comment]
Test for new sorting: 

```
irb(main):013:0> CourseAttendanceEvent.order_by_title.map { |x| x.attendance_event.title }
  CourseAttendanceEvent Load (1.3ms)  SELECT "course_attendance_events".* FROM "course_attendance_events"
  AttendanceEvent Load (1.1ms)  SELECT "attendance_events".* FROM "attendance_events" WHERE "attendance_events"."id" = $1 LIMIT $2  [["id", 1], ["LIMIT", 1]]
  AttendanceEvent Load (0.9ms)  SELECT "attendance_events".* FROM "attendance_events" WHERE "attendance_events"."id" = $1 LIMIT $2  [["id", 1], ["LIMIT", 1]]
  AttendanceEvent Load (1.2ms)  SELECT "attendance_events".* FROM "attendance_events" WHERE "attendance_events"."id" = $1 LIMIT $2  [["id", 2], ["LIMIT", 1]]
  AttendanceEvent Load (1.2ms)  SELECT "attendance_events".* FROM "attendance_events" WHERE "attendance_events"."id" = $1 LIMIT $2  [["id", 3], ["LIMIT", 1]]
  AttendanceEvent Load (1.8ms)  SELECT "attendance_events".* FROM "attendance_events" WHERE "attendance_events"."id" = $1 LIMIT $2  [["id", 4], ["LIMIT", 1]]
  AttendanceEvent Load (1.7ms)  SELECT "attendance_events".* FROM "attendance_events" WHERE "attendance_events"."id" = $1 LIMIT $2  [["id", 5], ["LIMIT", 1]]
  AttendanceEvent Load (1.1ms)  SELECT "attendance_events".* FROM "attendance_events" WHERE "attendance_events"."id" = $1 LIMIT $2  [["id", 6], ["LIMIT", 1]]
  AttendanceEvent Load (1.3ms)  SELECT "attendance_events".* FROM "attendance_events" WHERE "attendance_events"."id" = $1 LIMIT $2  [["id", 7], ["LIMIT", 1]]
  AttendanceEvent Load (1.8ms)  SELECT "attendance_events".* FROM "attendance_events" WHERE "attendance_events"."id" = $1 LIMIT $2  [["id", 8], ["LIMIT", 1]]
  AttendanceEvent Load (1.1ms)  SELECT "attendance_events".* FROM "attendance_events" WHERE "attendance_events"."id" = $1 LIMIT $2  [["id", 9], ["LIMIT", 1]]
  AttendanceEvent Load (1.4ms)  SELECT "attendance_events".* FROM "attendance_events" WHERE "attendance_events"."id" = $1 LIMIT $2  [["id", 10], ["LIMIT", 1]]
  AttendanceEvent Load (1.7ms)  SELECT "attendance_events".* FROM "attendance_events" WHERE "attendance_events"."id" = $1 LIMIT $2  [["id", 11], ["LIMIT", 1]]
=> ["A number that's not an LL: 1", "A number that's not an LL: 4", "A number that's not an LL: 21", "LC 1-1s", "Learning Lab 1: Onboard", "Learning Lab 2: Something", "Learning Lab 3: Third one", "Learning Lab 9: Another", "Learning Lab 10: Words", "Learning Lab 11: Another two", "test", "test"]
```

File: app/models/course_attendance_event.rb [hunk 2]
```diff
@@ -9,6 +9,18 @@ class CourseAttendanceEvent < ApplicationRecord
...
   validates :course, :attendance_event, :canvas_assignment_id, presence: true
 
+  # Sort names with numbers correctly.
+  # From https://stackoverflow.com/a/25042119/12432170.
+  scope :order_by_title, -> {
+    sort_by{ |e|
+      e.attendance_event.title.gsub(/\d+/) { |s|
+        # Left-pad numbers with zeroes.
+        # 8 is arbitrarily longer than the numbers we'll see in titles.
+        "%08d" % s.to_i
```
Comment (sadleb) [inline app/models/course_attendance_event.rb:19]: Huh, interesting.

**Pull request merged**